doc/security/implementation_guidelines/hardware/index.md

```diff
@@ -0,0 +1,409 @@
+8.  For maximum sensitivity, compare combinational and sequential paths with
+    hair-trigger/one-shot latch of miscompare
+9.  Empty detection for OTP/flash (if needed, but especially for lifecycle
+    determination)
+10. Avoid local resets / prefer larger reset domains, since a glitch on this
+    larger reset keeps more of the design "in sync." But, consider the
+    implications of any block with more than one reset domain (see also 9.1).
+11. Similar to the "mix in" idea of 4.4, in any case where multiple contributing
+    "votes" are going to an enable/unlock decision, consider mixing them into
+    some cryptographic structure over time that will be diverted from its path
+    by attempts to glitch each vote. (Note: if the final outcome of this is
+    simply a wide-compare that produces a single-bit final unlock/enable vote
+    then this is only marginally helpful - since that vote is now the glitch
+    target. Finding a way to bind the final cryptographic result to the vote is
+    preferred, but potentially very difficult / impossible, depending on the
+    situation.)
+12. When checking/creating a signal to
+    <span style="text-decoration:underline;">permit</span> some sensitive
+    operation, prefer that the checking logic is maximally volatile (e.g.
+    performs a lot of the calculation in a single cycle after a register), such
+    that a glitch prevents the operation. Whereas, when checking to
+    <span style="text-decoration:underline;">deny</span> a sensitive operation,
+    prefer that the checking logic is minimally volatile (is directly following
+    a register with minimal combinational logic), such that a glitch will be
+    recovered on the next clock and the denial will be continued/preserved.
+13. CFI (control flow integrity) hardware can help protect a processor /
+    programmable peripheral from some types of glitch attacks. This topic is
+    very involved and beyond the scope of these guidelines, consult [2] for an
+    introduction to previous techniques.
+14. Analog sensors (under/over-voltage, laser light, mesh breach, among others)
+    can be used to generate SoC-level alerts and/or inhibit sensitive
+    operations. Many of these sensors require calibration/trimming, or require
+    hysteresis circuits to prevent false-positives, so they may not be usable in
+    fast-reacting situations.
+15. Running an operation (e.g. AES or KMAC) to completion, even with a detected
+    fault, is sometimes useful since it suppresses information for the adversary
+    about the success/failure of the attempted fault, and minimizes any timing
+    side channel. However, for some operations (e.g. ECDSA sign), operations on
+    faulty inputs can have catastrophic consequences. These guidelines cannot
+    recommend a default-safe posture, but each decision about handling detected
+    faults should be carefully considered.
+16. Defending against SIFA, if considered in-scope, (see [15]/[16]) requires
```

Comment:
"if considered in-scope": I understand that Chris didn't necessarily have an answer to whether this was in scope when he drafted these guidelines, but I think it seems a bit silly for a project which an external reader will assume *has* a threat model to have text like this in its documentation.

Reply:
Removed.